common/build-style/zig-build.sh

```diff
@@ -0,0 +1,39 @@
+do_build() {
+	local zig_target zig_cpu
+
+	# TODO: This duplication between build-profiles cross-profiles is totally
+	# unnecessary. It would be nice if there was some way to avoid it.
+	if [ "$CROSS_BUILD" ]; then
+		zig_target="${XBPS_CROSS_ZIG_TARGET}"
+		zig_cpu="${XBPS_CROSS_ZIG_CPU}"
+	else
+		zig_target="${XBPS_ZIG_TARGET}"
+		zig_cpu="${XBPS_ZIG_CPU}"
+	fi
```

Comment:
What do you think about an `environment/build-style/` file that sets `archs` for the supported archs? This way you don't have to detect if these are set before `do_build` in order to trigger `broken` (or maybe it's the best way, since it avoids duplicating the information?).

Overall looks really good, just want to be sure of the best way of signaling when zig is unsupported.

Reply:
Hmm, that seems like it might be a good idea in general but I don't think we need that for zig. Zig is theoretically able to target all of xbps's targets and I've included exhaustive support for xbps's targets in this PR. 

However, support for some of the less common architectures is of course not as good and attempting to actually compile code for them may fail due to issues in the compiler and/or std. This of course depends on the exact code being compiled as well though, especially when the issues are due to stdlib deficiencies which are often easy to add temporary downstream workarounds for as the language matures.